components/camel-jt400/src/main/java/org/apache/camel/component/jt400/Jt400MsgQueueProducer.java

```diff
@@ -50,7 +50,12 @@ public void process(Exchange exchange) throws Exception {
 
     private void process(MessageQueue queue, Exchange exchange) throws Exception {
         String msgText = exchange.getIn().getBody(String.class);
-        queue.sendInformational(msgText);
+        byte[] messageKey = (byte[]) exchange.getIn().getHeader(Jt400Constants.MESSAGE_REPLYTO_KEY);
+        if (null != messageKey && !msgText.isEmpty()) {
```

Comment:
For better readability here, I would think this may make more sense:
```suggestion
        if (ObjectHelper.isNotEmpty(messageKey) && ObjectHelper.isNotEmpty(msgText)) {
```

Whereby `ObjectHelper` is from `org.apache.camel.util.ObjectHelper`